src/dst/src/LoadBalancer.cc

```diff
@@ -103,6 +105,46 @@ void LoadBalancer::getNextWorker(ip::address& ip, unsigned short& port)
   }
 }
 
+void LoadBalancer::lookUpWorkers(const char* domain, unsigned short port)// LoadBalancer & balancer)
+{
+  asio::io_service ios;
+  udp::resolver::query resolver_query(domain, std::to_string(port), udp::resolver::query::numeric_service);
+  udp::resolver resolver(ios);
+  while(true){
+    std::vector<worker> new_workers;
+    boost::system::error_code ec;
+    auto it = resolver.resolve(resolver_query, ec);
+    if(ec)
+      logger_->error(utl::DST, 203, "Workers domain resolution failed with error code = {}. Message = {}.", 
+        ec.value(),
+        ec.message());
+    int new_workers_count = 0;
+    udp::resolver::iterator it_end;
+    for (; it != it_end; ++it) {
+        auto discovered_worker = worker(it->endpoint().address(), port, 0);
+        if(std::find(workers_set.begin(), workers_set.end(), discovered_worker) == workers_set.end()){
+          workers_set.push_back(discovered_worker);
```

Comment:
I think workers_set could be local variable to the function only since it'not used anywhere else and lookupWorkers doesn't return.